templates/element/request_panel.php

```diff
@@ -34,6 +34,9 @@
 <h4><?= __d('debug_kit', 'Routing Params') ?></h4>
 <?= $this->Toolbar->dump($params) ?>
 
+<h4><?= __d('debug_kit', 'Attributes') ?></h4>
+<?= $this->Toolbar->dump($attributes); ?>
```

Comment:
Should we remove `Routing Params` instead of having duplicate data?

Reply:
not sure how users feel about `Routing Params` , and they know it is in `attributes` list or no

Reply:
That's fair, but if it is the only params list in the panel they will figure it out. We will be leaking some implementation details, but those details are already a public interface. The `getParam()` methods are ease of use methods as the required existence checks to safely use the return of `getAttribute()` are tedious.

Reply:
removed

Reply:
`getAttributes()` always have at least 4 items :point_down:  

https://github.com/cakephp/cakephp/blob/41ce80827fd964f6b32e799c837fc3c585a097c1/src/Http/ServerRequest.php#L1674-L1679

Reply:
added